### ISSUE TITLE
Icons not showing up, issue related to devicons dependency, + Octicon support

### ISSUE THREAD

**Issue by rogeruiz** [MadcapJake/fly-notify#4]
From what I see [here](https://github.com/MadcapJake/fly-notify/blob/master/index.js#L22-L26), the `devicons` function is culling the file from the `!SVG` directory in the `devicons`. I tested using `terminal-notify` with SVG icons and had no luck. I also tried the `!PNG` directory instead, and the icons were very blurry on my display. I wrote up a script that'll convert the `devicons` SVG files to `450x450` PNGs using ImageMagick's `convert`, but since it's a dependency, I wasn't sure of the best way to include this in the project. 

I'd also like to add the Octicon's mentioned in the Readme too. That'd be rad! ✌️

**Comment by MadcapJake** [comment]
Yeah the pngs were blurry for me but svgs worked OOTB for me.  I don't follow what you mean by `terminal-notify`. I wonder if svgs working is an OS-dependent thing.

**Comment by rogeruiz** [comment]
Hmmmm. Sounds like it.

I'm on Mac OS X 10.10.3. So maybe it's just on a Mac that they don't show up properly.

Sorry I didn't link to [this](https://github.com/mikaelbr/node-notifier#thanks-to-oss) earlier, `terminal-notifier` is the module that `node-notifier` will use on Mac OS X 10.8+.

We could leverage the [notifiers here](https://github.com/mikaelbr/node-notifier/tree/master/notifiers) to set up some custom options. Here's a look at [the docs](https://github.com/mikaelbr/node-notifier#cross-platform-advanced-usage) – you'll need to scroll down a bit. That way we don't even serve the SVGs to Mac or any other platforms that don't support them.

As for the conversion, here's the sketch of what I wrote up to convert everything from SVG to PNG.

``` sh
#!/bin/sh

cd ./\!SVG/

for file in ./*.svg
do
  if [ -f "$file" ]; then
    name=$(echo $file | grep -o '[^\.\/].\+[^\.svg]')
    convert -density 4000 -background transparent "$file" -resize 450x450 ../\!PNG/"${name}.png"
  fi
done
```

I'd hate to add ImageMagick as a project dependency, so I could just do this locally and check it in for now or make it a dev dependency and include a script. I [don't see these icons updating too often](https://github.com/vorillaz/devicons/commits/master), but I would hate to see Issues open up here for icons that were added to the `devicons` project. ¯_(ツ)_/¯

As for the Octicons. Was there anything in particular that you were planning for that? They're [all there in the `svg/` directory](https://github.com/github/octicons/tree/master/svg) so converting them like the devicons would be a snap.

I'll look into the SVG/PNG issues across platforms and see what I can do to test a bit more using the notifier's options for Mac OS X 10.8+ available in node-notifiers.

**Comment by MadcapJake** [comment]
Yeah I looked through `node-notifer` and it makes no mention of whether svgs would work.  Looking over `terminal-notifier`, I can't find anything either but I don't really have any experience with Objective C.  I can't decide what the best route is for this plugin:
- Commit a converted set of icons. This would need to be updated whenever oct/dev icons updates.
- Create a separate OSX version by finding out the os and running the NotificationCenter version of `node-notifer`.  This might work but would add some os-specific code to `index.js` which I can't debug (Linux user). However, this might mean we could avoid doing any image conversion.
- Maybe just convert each icon on the fly (no pun intended).  A function would take the `icon` string, check the OS, if `OSX` it would convert to png and serve that, otherwise it just serves SVG.  This could be cleaner or it could just be unecessary complexity.
- Another option (that I think I'm leaning towards) is to have the image conversion happen `postinstall`.  This way, npm could download the latest devicons/octicons and gather and convert into `fly-notify`'s dir.  Users will always get the latest version of icons and `index.js` will stay really concise.

It would be helpful if we had a Windows user too.  If it doesn't work on Windows with SVG, then it might make more sense to do a more heavy-handed approach.
